src/custom-element.js

```diff
@@ -668,21 +701,58 @@ function createBaseAmpElementProto(win) {
         }
         this.implementation_.layout_ = this.layout_;
         this.implementation_.firstAttachedCallback();
-        if (!isUpgraded(this)) {
-          // amp:attached is dispatched from the ElementStub class when it replayed
-          // the firstAttachedCallback call.
+        if (!this.isUpgraded()) {
+          // amp:attached is dispatched from the ElementStub class when it
+          // replayed the firstAttachedCallback call.
           this.dispatchCustomEvent('amp:stubbed');
```

Comment:
I found event here could be element not stubbed but fail to post upgrade? which direct to  upgrade status `UPGRADE FAILED` or upgraded state `NOT UPGRADED`.